lib/ops/opsservice/shrink.go

```diff
@@ -406,6 +410,23 @@ func (s *site) shrinkOperationStart(ctx *operationContext) (err error) {
 	return nil
 }
 
+func (s *site) pickShrinkMasterRunner(ctx *operationContext, removedServer storage.Server) (*serverRunner, error) {
+	masters, err := s.getTeleportServers(schema.ServiceLabelRole, string(schema.ServiceRoleMaster))
```

Comment:
This is already done in `validateShrinkRequest` - I wonder if you can reuse the results (it is also better to init in `validateShrinkRequest` **before** starting the operation) and set the masterRunner as well as initialize teleport server for the server being removed.

Reply:
Server can't be initialized there b/c "validate" and "start" functions can be executed by different gravity-sites (if the removed node is active master) so operation context is not passed directly from "validate" to "start". I guess I could determine master server in "validate" and save it in the operation data so "start" would later fetch and initialize it, but I'm not sure how that's better?

Reply:
ok, good point.